### PR TITLE
Add platform admin page for uuid queries

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

Add a page to allow queries of arbitrary uuids (notifications/services/templates/users/jobs) and get the relevant links.

image

# Test instructions | Instructions pour tester la modification

- Run this locally connected to the staging api or use the [review app](https://kxitqumga2jjuzybsfol5rncti0slyhf.lambda-url.ca-central-1.on.aws/).
- Go to the platform admin / Search for ids page and search away.

### CODE REVIEW

File: app/config.py [hunk 1]
```diff
@@ -25,8 +25,7 @@ class Config(object):
     ALLOW_DEBUG_ROUTE = env.bool("ALLOW_DEBUG_ROUTE", False)
 
     # List of allowed service IDs that are allowed to send HTML through their templates.
-    ALLOW_HTML_SERVICE_IDS: List[str] = [
-        id.strip() for id in os.getenv("ALLOW_HTML_SERVICE_IDS", "").split(",")]
```
Comment (sastels) [inline app/config.py:29]: All the changes in this file were done by `make format`.

File: app/utils.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ def from_lambda_api(line):
     return isinstance(line, dict)
 
 
-@cache.memoize(timeout=12*60*60)
+@cache.memoize(timeout=12 * 60 * 60)
```
Comment (sastels) [inline app/utils.py:96]: `make format`

File: app/__init__.py [hunk 3]
```diff
@@ -165,6 +166,7 @@ def get_locale():
         provider_client,
```
[the remark this answers is not in the document]
Comment (jimleroyer) [reply app/__init__.py:166]: It is in the right order. 🤖

File: app/main/forms.py [hunk 4]
```diff
@@ -1329,6 +1329,13 @@ class SearchUsersByEmailForm(StripWhitespaceForm):
     )
 
 
+class SearchIds(StripWhitespaceForm):
+    search = SearchField(
+        _l("List of IDs"),
+        validators=[DataRequired(_l("You need to enter one or more UUIDs to search by."))],
```
[the remark this answers is not in the document]
Comment (jimleroyer) [reply app/main/forms.py:1335]: That won't make much difference.

File: app/notify_client/support_api_client.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        self.contact_email = app.config["CONTACT_EMAIL"]
+        self.notify_user_id = app.config["NOTIFY_USER_ID"]
+        self.notify_service_id = app.config["NOTIFY_SERVICE_ID"]
+
+    def find_ids(self, ids):
+        data = self.get("/support/find-ids", params={"ids": ids})
+        return data
+
+
+support_api_client = SupportApiClient()
```
[the remark this answers is not in the document]
Comment (jimleroyer) [reply app/notify_client/support_api_client.py:17]: Not a bad recommendation but not the pattern that the app is following.

File: app/main/views/find_ids.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+from app.utils import user_is_platform_admin
+
+
+@main.route("/find-ids", methods=["GET", "POST"])
+@user_is_platform_admin
+def find_ids():
+    form = SearchIds()
+    records = None
+    if form.validate_on_submit():
+        records = support_api_client.find_ids(form.search.data)
```
[the remark this answers is not in the document]
Comment (sastels) [reply app/main/views/find_ids.py:15]: A generally good idea, but in this case if there's a server error with api then it's fine for the admin web page to show an error.

File: app/notify_client/support_api_client.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+class SupportApiClient(NotifyAdminAPIClient):
+    def init_app(self, app):
+        super().init_app(app)
+        self.admin_url = app.config["ADMIN_BASE_URL"]
+        self.contact_email = app.config["CONTACT_EMAIL"]
+        self.notify_user_id = app.config["NOTIFY_USER_ID"]
+        self.notify_service_id = app.config["NOTIFY_SERVICE_ID"]
+
+    def find_ids(self, ids):
+        data = self.get("/support/find-ids", params={"ids": ids})
```
[the remark this answers is not in the document]
Comment (sastels) [reply app/notify_client/support_api_client.py:13]: nah, just show the error in the web app.

File: app/templates/views/find-ids/find-ids.html [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    <div class="md:w-full float-left py-0 px-0 px-gutterHalf box-border">
+      {{ textbox_search(
+        form.search,
+        width='w-full',
+        label=label_txt
+      ) }}
+    </div>
+  {% endcall %}
+
+  {% call form_wrapper(id='search-form' ) %}
```
[the remark this answers is not in the document]
Comment (sastels) [reply app/templates/views/find-ids/find-ids.html:31]: 😬 This is just a copy of how we do similar searches elsewhere so I think I'll leave it alone.

**Review by jimleroyer — APPROVED** [review]
LGTM :shipit:

**Pull request merged**